src/NServiceBus.Testing/TestingContext.cs

```diff
@@ -12,31 +10,14 @@ public TestingContext(IMessageCreator messageCreator = null) : base(messageCreat
 
         public TestingContext(IMessageCreator messageCreator, TimeoutMessage<object>[] timeoutMessages) : base(messageCreator)
         {
-            this.timeoutMessages = new ConcurrentQueue<TimeoutMessage<object>>(timeoutMessages);
+            this.previousTimeouts = timeoutMessages;
         }
 
-        public TimeoutMessage<object>[] TimeoutMessages => timeoutMessages.ToArray();
```

Comment:
but we need this special code in here to track timeouts across multiple contexts which is very specific to how the test framework works. When you're testing sagas as unit tests, this code here isn't required and the `TimeoutMessages` on the pipelinecontext is good enough.